yroots/subdivision.py

```diff
@@ -335,16 +340,28 @@ def interval_approximate_nd(f,a,b,deg,return_bools=False,inf_norm=None):
     dim = len(a)
 
     if hasattr(f,"evaluate_grid"):
-        xyz = transform(get_cheb_grid(deg, dim, True), a, b)
-        values_block = f.evaluate_grid(xyz)
+        cheb_points = transform(get_cheb_grid(deg, dim, True), a, b)
+        values_block = f.evaluate_grid(cheb_points)
     else:
         cheb_points = transform(get_cheb_grid(deg, dim, False), a, b)
         cheb_points = [cheb_points[:,i] for i in range(dim)]
         values_block = f(*cheb_points).reshape(*([deg+1]*dim))
 
     # figure out on which subintervals the function changes sign
     if return_bools:
-        change_sign = np.zeros(2**dim, dtype=bool)
+        change_sign = [False]*(2**dim)
```

Comment:
Thanks for looking into this. With the faster implementation above, can you try this again? If at the end of the it's still slower, let's just take out the change_sign everywhere in the code for cleanliness.